packages/ui-kit/src/components/AccessTokenProvider/AccessTokenProvider.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { createContext, PropsWithChildren, useCallback, useEffect, useRef, useState } from 'react'
+import { useLog } from '../Log'
+
+export interface AccessTokenContextValue {
+  /**
+   * The returned accessToken wheter from the props or from the fetchToken function
+   */
+  accessToken?: string
+  /**
+   * If true, the access token is being fetched
+   */
+  isLoading?: boolean
```

Comment:
What would the customer do with this?

I also tend to think we should avoid introducing this, at least in the first iteration. Instead, why not log at debug?

Reply:
The components use this variable in order to know whether the access token is being fetched or not to show the loader.

Again, I think this one would be more something internal, I think it's worth mentioning here that neither `failedRetry` or `isLoading` are props, those are just states that the provider uses internally.